components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5897,8 +5915,14 @@ where
         }
         self.fsm.peer.size_diff_hint = 0;
         self.fsm.peer.compaction_declined_bytes = 0;
-        // the task is scheduled, next tick may skip it.
-        self.fsm.peer.may_skip_split_check = true;
+        // the task is scheduled, next tick may skip it only when the size and keys are
+        // small If either size or keys are big enough to do a split,
+        // keep split check tick until split is done
+        if !matches!(self.fsm.peer.may_split_size, Some(true))
+            && !matches!(self.fsm.peer.may_split_keys, Some(true))
```

Comment:
How about
```suggestion
        if !matches!(self.fsm.peer.maybe_splittable, Some(true))
```

Reply:
There's a problem with single maybe_splittable. Because what if keys checker says it's not splitable, but size check says it's splitable? The maybe_splittable could be incorrectly overwritten by either of the value (though today the caller use same splitable value, it's possible in the future it can be different).  So having two variables is more robust.